src/cuga/backend/cuga_graph/policy/tool_guard/tool_invoker.py

```
@@ -0,0 +1,123 @@
+            result = await tool.ainvoke(arguments)
+            
+            logger.debug(f"Tool '{toolname}' invocation successful")
+            return result
+            
+        except ValueError:
+            # Re-raise ValueError as-is (tool not found)
+            raise
+        except Exception as e:
+            logger.error(f"Failed to invoke tool '{toolname}': {e}")
+            raise RuntimeError(
+                f"Tool invocation failed for '{toolname}': {str(e)}"
```

Ruff (0.15.12)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)